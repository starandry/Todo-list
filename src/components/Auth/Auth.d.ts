import React from 'react'
import { User } from 'firebase/auth'
declare const Auth: React.FC<{
    onLogin: (user: User) => void
}>
export default Auth
