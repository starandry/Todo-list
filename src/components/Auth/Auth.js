import { jsx as _jsx, jsxs as _jsxs } from 'react/jsx-runtime'
import { useState } from 'react'
import { signInWithEmailAndPassword, createUserWithEmailAndPassword } from 'firebase/auth' // Импортируем тип User
import { auth } from '../../../firebaseConfig'
import styles from './Auth.module.scss'
const Auth = ({ onLogin }) => {
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [isLogin, setIsLogin] = useState(true)
    const [error, setError] = useState(null)
    const handleSubmit = async (e) => {
        e.preventDefault()
        setError(null)
        try {
            if (isLogin) {
                // Вход
                const userCredential = await signInWithEmailAndPassword(auth, email, password)
                onLogin(userCredential.user)
            } else {
                // Регистрация
                const userCredential = await createUserWithEmailAndPassword(auth, email, password)
                onLogin(userCredential.user)
            }
        } catch (err) {
            setError(err.message)
        }
    }
    return _jsx('div', {
        className: styles['auth-container'],
        children: _jsxs('div', {
            className: styles['auth-box'],
            children: [
                _jsx('h2', { children: isLogin ? 'Login' : 'Register' }),
                _jsxs('form', {
                    onSubmit: handleSubmit,
                    children: [
                        _jsx('input', {
                            type: 'email',
                            placeholder: 'Email',
                            value: email,
                            onChange: (e) => setEmail(e.target.value),
                        }),
                        _jsx('input', {
                            type: 'password',
                            placeholder: 'Password',
                            value: password,
                            onChange: (e) => setPassword(e.target.value),
                        }),
                        _jsx('button', { type: 'submit', children: isLogin ? 'Login' : 'Register' }),
                    ],
                }),
                error && _jsx('p', { children: error }),
                _jsx('button', {
                    className: styles['switch-btn'],
                    onClick: () => setIsLogin(!isLogin),
                    children: isLogin ? 'Switch to Register' : 'Switch to Login',
                }),
            ],
        }),
    })
}
export default Auth
