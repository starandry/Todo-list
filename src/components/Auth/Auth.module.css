.auth-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    background-color: #f0f0f5;
}

.auth-box {
    background: #ffffff;
    padding: 40px;
    border-radius: 10px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    width: 350px;
    text-align: center;
}
.auth-box h2 {
    margin-bottom: 20px;
    font-size: 1.5rem;
    color: #333;
}
.auth-box form {
    display: flex;
    flex-direction: column;
    gap: 15px;
}
.auth-box input {
    padding: 10px;
    border-radius: 5px;
    border: 1px solid #ccc;
    font-size: 1rem;
    width: 100%;
    outline: none;
}
.auth-box button {
    padding: 10px 15px;
    background-color: #28a745;
    color: #fff;
    border: none;
    border-radius: 5px;
    font-size: 1rem;
    cursor: pointer;
    transition: background-color 0.3s ease;
}
.auth-box button:hover {
    background-color: #218838;
}
.auth-box .switch-btn {
    background-color: #007bff;
    margin-top: 15px;
}
.auth-box .switch-btn:hover {
    background-color: #0056b3;
}
.auth-box p {
    color: red;
    font-size: 0.9rem;
}

/*# sourceMappingURL=Auth.module.css.map */
