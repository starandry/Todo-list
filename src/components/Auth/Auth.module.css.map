{"version":3,"sourceRoot":"","sources":["Auth.module.scss"],"names":[],"mappings":"AAAA;EACI;EACA;EACA;EACA;EACA;;;AAGJ;EACI;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;EACA;;AAGJ;EACI;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;;AAIR;EACI;EACA;;AAEA;EACI;;AAIR;EACI;EACA","file":"Auth.module.css"}