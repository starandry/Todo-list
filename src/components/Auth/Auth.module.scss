.auth-container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    background-color: #f0f0f5; // Светлый фон
}

.auth-box {
    background: #ffffff;
    padding: 40px;
    border-radius: 10px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    width: 350px;
    text-align: center;

    h2 {
        margin-bottom: 20px;
        font-size: 1.5rem;
        color: #333;
    }

    form {
        display: flex;
        flex-direction: column;
        gap: 15px;
    }

    input {
        padding: 10px;
        border-radius: 5px;
        border: 1px solid #ccc;
        font-size: 1rem;
        width: 100%;
        outline: none;
    }

    button {
        padding: 10px 15px;
        background-color: #28a745;
        color: #fff;
        border: none;
        border-radius: 5px;
        font-size: 1rem;
        cursor: pointer;
        transition: background-color 0.3s ease;

        &:hover {
            background-color: #218838;
        }
    }

    .switch-btn {
        background-color: #007bff;
        margin-top: 15px;

        &:hover {
            background-color: #0056b3;
        }
    }

    p {
        color: red;
        font-size: 0.9rem;
    }
}
