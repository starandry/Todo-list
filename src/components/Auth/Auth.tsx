import React, { useState } from 'react'
import { signInWithEmailAndPassword, createUserWithEmailAndPassword, User } from 'firebase/auth' // Импортируем тип User
import { auth } from '../../../firebaseConfig'
import styles from './Auth.module.scss' // Импортируем стили

const Auth: React.FC<{
    onLogin: (user: User) => void
}> = ({ onLogin }) => {
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [isLogin, setIsLogin] = useState(true)
    const [error, setError] = useState<string | null>(null)

    const handleSubmit = async (e: React.FormEvent) => {
        e.preventDefault()
        setError(null)

        try {
            if (isLogin) {
                // Вход
                const userCredential = await signInWithEmailAndPassword(auth, email, password)
                onLogin(userCredential.user) // user возвращает тип User
            } else {
                // Регистрация
                const userCredential = await createUserWithEmailAndPassword(auth, email, password)
                onLogin(userCredential.user) // user возвращает тип User
            }
        } catch (err) {
            setError((err as Error).message)
        }
    }

    return (
        <div className={styles['auth-container']}>
            <div className={styles['auth-box']}>
                <h2>{isLogin ? 'Login' : 'Register'}</h2>
                <form onSubmit={handleSubmit}>
                    <input type="email" placeholder="Email" value={email} onChange={(e) => setEmail(e.target.value)} />
                    <input
                        type="password"
                        placeholder="Password"
                        value={password}
                        onChange={(e) => setPassword(e.target.value)}
                    />
                    <button type="submit">{isLogin ? 'Login' : 'Register'}</button>
                </form>
                {error && <p>{error}</p>}
                <button className={styles['switch-btn']} onClick={() => setIsLogin(!isLogin)}>
                    {isLogin ? 'Switch to Register' : 'Switch to Login'}
                </button>
            </div>
        </div>
    )
}

export default Auth
