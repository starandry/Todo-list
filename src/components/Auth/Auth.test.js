import { jsx as _jsx } from 'react/jsx-runtime'
import { render } from '@testing-library/react'
import Auth from './Auth'
describe('Auth Component', () => {
    it('renders correctly', () => {
        const mockOnLogin = jest.fn(() => {})
        const { asFragment } = render(_jsx(Auth, { onLogin: mockOnLogin }))
        expect(asFragment()).toMatchSnapshot()
    })
})
