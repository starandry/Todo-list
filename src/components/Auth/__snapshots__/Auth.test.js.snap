// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="auth-container"
  >
    <div
      class="auth-box"
    >
      <h2>
        Login
      </h2>
      <form>
        <input
          placeholder="Email"
          type="email"
          value=""
        />
        <input
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Login
        </button>
      </form>
      <button
        class="switch-btn"
      >
        Switch to Register
      </button>
    </div>
  </div>
</DocumentFragment>
`;
